### ISSUE TITLE
ktor client does not log request body

### ISSUE THREAD

**Issue by AlexCzar** [ktorio/ktor#1729]
**Ktor Version and Engine Used (client or server and name)**
ver: 1.3.1, 1.3.2
engine: Apache
app: client
dependencies:
```kotlin
dependencies {
    implementation(kotlin("stdlib-jdk8"))
    implementation(platform("io.ktor:ktor-bom:1.3.2"))
    implementation("io.ktor:ktor-client-core")
    implementation("io.ktor:ktor-client-logging-jvm")
    implementation("io.ktor:ktor-client-apache")
    implementation("io.ktor:ktor-client-json-jvm")
    implementation("org.slf4j:slf4j-simple:1.7.26")
    implementation("org.apache.httpcomponents:httpasyncclient:4.1.4")
}
```

**Describe the bug**
Request body is not logged even with LogLevel.ALL

**To Reproduce**
```kotlin
package com.example

import io.ktor.client.HttpClient
import io.ktor.client.engine.apache.Apache
import io.ktor.client.features.logging.DEFAULT
import io.ktor.client.features.logging.LogLevel
import io.ktor.client.features.logging.Logger
import io.ktor.client.features.logging.Logging
import io.ktor.client.request.post
import io.ktor.util.KtorExperimentalAPI
import kotlinx.coroutines.runBlocking

@KtorExperimentalAPI
fun main() {
    runBlocking {
        val client = HttpClient(Apache) {
            install(Logging) {
                logger = Logger.DEFAULT
                level = LogLevel.ALL
            }
        }
        client.post<String>("http://localhost:8080") {
            body = """{ "test": "testVal" }"""
        }
    }
}
```

**Expected behavior**
Request body is in the log

**Actual result**
```text
[main] INFO io.ktor.client.HttpClient - REQUEST: http://localhost:8080/
[main] INFO io.ktor.client.HttpClient - METHOD: HttpMethod(value=POST)
[main] INFO io.ktor.client.HttpClient - COMMON HEADERS
[main] INFO io.ktor.client.HttpClient - -> Accept-Charset: UTF-8
[main] INFO io.ktor.client.HttpClient - -> Accept: */*
[main] INFO io.ktor.client.HttpClient - CONTENT HEADERS
[main] INFO io.ktor.client.HttpClient - BODY Content-Type: text/plain; charset=UTF-8
[main] INFO io.ktor.client.HttpClient - RESPONSE: 200 
---   remaining response stuff skipped for brevity  ---
```

**Comment by geoffreywiseman** [comment]
Yes, just hit this -- I'm getting a validation error from the server, and since Ktor is doing the serialization, would be helpful to be able to compare the request and response. Solved it by using a proxy, but configuring an HTTPS proxy is much more annoying than logging would have been.

**Comment by huehnerlady** [comment]
Any update on this? We run into the same issue here and seeing the request body would help a lot

**Comment by weickmanna** [comment]
One advantage of ktor is that it is extremely modular and therefore can be customized to one's one needs rather easily.

We found the ktor client logging not fit our needs and since it is also not really configurable, we wrote our own custom client logging feature. You can just copy and paste from the original Logging and adapt as needed.

Not saying that this bug shouldn't be fixed in ktor if it already offers this option. But if it is urgent for you, you might have success by rolling your own implementation and that might be more light weight than having a separate proxy.

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-442">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed in 1.4.1
